### DISCUSSION TITLE
Realtime safety requirements for the logging extension

### DISCUSSION THREAD

**Discussion by robbert-vdh** [free-audio/clap#185] [Q&A] [answered]
The logging extension mentions that the logging callbacks must be callable from any thread: https://github.com/free-audio/clap/blob/main/include/clap/ext/log.h My question is, should the plugin also assume that calling these from a process function is realtime safe? (for as far as logging can be realtime safe) I assume not, but I thought I'd get some clarification first. I was considering how to support the extension in [yabridge](https://github.com/robbert-vdh/yabridge), and the options are to either a) not support it at all, b) proxy the callbacks as normal (meaning that there will be an allocation for an `std::string` and a back and forth function call over a a socket, or c) only allocate the string and defer sending the actual callback (would make using these functions from a process function slightly safer, but it still needs to allocate, also a lot more complicated). If there are no realtime safety requirements I'll just keep things simple and go with b).

**Comment by abique** [comment, marked as answer]
The host should be able to deal with it up to some extents, in worst case the host simply drops the log message.

**Reply by abique** [reply]
I think it is possible to have some memory available for logging message, but of course this memory isn't infinite, and if the plugin outputs a lot, it'll fill the buffer quickly, maybe quicker than the main thread could pump it due to the scheduler settings.

**Reply by abique** [reply]
I would not log from the audio thread, unless there is a **very** important message like critical error message. Because there will always be some host implementing this logging feature in a non real-time way.
